Feature/issue#7

ErikBetzholtz opened the pull request:
This merge will close issue #7 by adding the function LIC3 to the Main class for later usage in DECIDE function. New file containing tests for LIC3 and future LIC functions is also included

ErikBetzholtz committed:
feat: add method for LIC3

This commit will add a method for LIC3 in the Main class for later integration into the DECIDE method. It will also create a new file containing tests for LIC3 and is intended for future tests of other LIC methods

ErikBetzholtz committed:
Squashed commit of the following:

commit a2feac7e7cb04018fae1402ee779cd1d09f92b4d
Merge: 7c9c0b7 1cf1d9b
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Mon Jan 27 19:48:51 2025 +0100

    Merge pull request #23 from VincentLagerros/globalDeclerations

    feat: add global declarations according to the assignment specificati…

commit 1cf1d9b14f4a0624446d5948ad016d7c87d5f3ef
Author: Julia Potrus <juliapo@kth.se>
Date:   Mon Jan 27 18:42:06 2025 +0100

    fix: open Globals constructor to avoid problems when testing

commit d5c832f382d81d7ebf68289c5e277772716e82e0
Author: Julia Potrus <juliapo@kth.se>
Date:   Mon Jan 27 17:17:36 2025 +0100

    bug: add reset function for singelton, making tests independent

ErikBetzholtz committed:
chore: add in-code documentation for LIC3

This commit will add documentation for the function LIC3

RasmussBrodin commented:
Nice work!

The code seems to be working as intended and the tests are relevant. I'm unsure if we should write the methods for LIC in main, however. I think that it is more appropriate to create a new class where we have all the methods for LIC?

Since the code works it is not super important but I think it would be easier to initialize and update the points in the loop and change the loop-guard to remove the if statement in the loop. Something like this:

for (int i = 0; i < xCoordinates.length - 2; i++) {
        // Get the coordinates of the three points
        double x1 = xCoordinates[i];
        double y1 = yCoordinates[i];
        double x2 = xCoordinates[i + 1];
        double y2 = yCoordinates[i + 1];
        double x3 = xCoordinates[i + 2];
        double y3 = yCoordinates[i + 2];

ErikBetzholtz committed:
refactor: change LIC3 for better readability and modes from main to new class

This commit will refactor the LIC3 by making it more compact and with coordinate variables being assigned values only inside the loop. It will also move LIC3 to new class EvaluateLIC in new file with same name

ErikBetzholtz committed:
doc: add short description for LIC3 test cases

This commit will add short descriptions of the three test cases related to the LIC3 function

ErikBetzholtz commented:
I will move the function to the new EvaluateLIC class. Also I don't know why i did not do the loop like that to begin with haha. Working on it right now

ErikBetzholtz committed:
Squashed commit of the following:

commit 5f46adb78b4bc40e5c667eecf087dc2f0483cd60
Merge: f748782 86f42df
Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
Date:   Tue Jan 28 13:36:08 2025 +0100

    Merge pull request #25 from VincentLagerros/feature/issue#10

    Feature/issue#10 Implement LIC6 method and tests

commit 86f42dfa363a7f77cdfb6bae45beb3ccfd1bfd95
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Tue Jan 28 13:29:17 2025 +0100

    fix: Updated comment explaining function

    Added missing text explaining the function of the method.

commit 6947f8532f6ab3c8ffc96b5e1a0f0af312dc2569
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Tue Jan 28 13:24:45 2025 +0100

    Fix: Added edge test cases and changed name of test file

    Added test case for when all points are on the same line, and when nPts=numPoints=3.

    Changed name of test file to EvaluateLICTest to conform with convention.

commit f748782cddac9ac51434a8c52bd4f25ac6438978
Merge: a2feac7 532804f
Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
Date:   Tue Jan 28 12:29:05 2025 +0100

    Merge pull request #26 from VincentLagerros/readme

    Doc: add README according to template

commit 62f218c09cc466bcd177badfcaaa7caef69f2013
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Tue Jan 28 12:16:12 2025 +0100

    Changed name of LIC test class

    Changed the name of the LIC test class so all tests relating to LIC can be placed there.

commit 532804fde2977dfd747a4f89e9d6cb71e109d05b
Author: Julia Potrus <juliapo@kth.se>
Date:   Tue Jan 28 11:51:58 2025 +0100

    Doc: add info that licesce is included in directory

commit cf8d15bdf716fbb09426f5e9a185b11dd76c73f1
Author: Julia Potrus <juliapo@kth.se>
Date:   Tue Jan 28 11:49:26 2025 +0100

    add license

commit e5f911853cc466d9fcd43d80e23f679adb0eb08d
Author: Julia Potrus <juliapo@kth.se>
Date:   Tue Jan 28 11:38:58 2025 +0100

    Update name of the course

commit 568930c3ecd1d74d32c4e7fd4b594babdfb6ef9f
Author: Julia Potrus <juliapo@kth.se>
Date:   Tue Jan 28 11:28:01 2025 +0100

    add README according to template

commit 9c77b5160669c4f8a7a04df48ff5b7f1c4789ff4
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Tue Jan 28 10:02:56 2025 +0100

    Feat: Small update for test testLIC6NeagtiveShort

    Fixed incorrect parameter in test where numPoints should be less than 3.

commit 3cbc62ff0532d05905d7d6735610a213577885a1
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Tue Jan 28 10:01:01 2025 +0100

    Feat: Added class for LIC and method for LIC6

    This commit will add a class for all LIC  methods and a method for LIC6 for later integration into the DECIDE method.

    It will also create a new file containing tests for LIC6.

commit a2feac7e7cb04018fae1402ee779cd1d09f92b4d
Merge: 7c9c0b7 1cf1d9b
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Mon Jan 27 19:48:51 2025 +0100

    Merge pull request #23 from VincentLagerros/globalDeclerations

    feat: add global declarations according to the assignment specificati…

commit 1cf1d9b14f4a0624446d5948ad016d7c87d5f3ef
Author: Julia Potrus <juliapo@kth.se>
Date:   Mon Jan 27 18:42:06 2025 +0100

    fix: open Globals constructor to avoid problems when testing

commit d5c832f382d81d7ebf68289c5e277772716e82e0
Author: Julia Potrus <juliapo@kth.se>
Date:   Mon Jan 27 17:17:36 2025 +0100

    bug: add reset function for singelton, making tests independent

ErikBetzholtz committed:
refactor: move content from LICTest to EvaluateLICTest and move LIC3 to EvaluateLIC

This commit will move tests for LIC from the deleted LICTest to EvaluateLICTest and LIC3 from Main to EvaluateLIC. This is to clean up after local merge with main

ErikBetzholtz committed:
Merge branch 'main' of https://github.com/VincentLagerros/DD2480-Group-5 into feature/issue#7

RasmussBrodin commented:
Looks great!
